### PR TITLE
Annotations loading categories

### PR DESCRIPTION
**Pull request by libby-natola**
fixed that error reporting issue caused by having the datasetsAnnotationsByCategory object within the state.api.errors object

### CODE REVIEW

File: src/store/store.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ api.namespaced = true;
 
 export default new Vuex.Store({
   state() {
-    return { title: null, filterSummary: null, filterDescription: null, debug: debug, lastError: null };
+    return { title: null, filterSummary: null, filterDescription: null, debug: debug, lastError: null, datasetsAnnotationsLoading: {} };
```
Comment (arteymix) [inline src/store/store.js:15]: This line is not necessary, the state is stored in the `api` module.

File: src/store/store.js [hunk 2]
```diff
@@ -29,6 +29,9 @@ export default new Vuex.Store({
     },
     setLastError(state, newVal) {
       state.lastError = newVal;
+    },
+    SET_ANNOTATIONS_LOADING(state, { category, loading }) {
+      Vue.set(state.datasetsAnnotationsLoading, category, loading);
```
Comment (arteymix) [inline src/store/store.js:34]: This is unnecessary too.

File: src/views/Browser.vue [hunk 3]
```diff
@@ -325,10 +325,11 @@ export default {
         } else {
           return Object.values(state.api.error)
             .filter(e => e !== null)
+            .filter(e => e !== state.api.error.datasetsAnnotationsByCategory || !Object.values(e).every(value => value === null))
```
Comment (arteymix) [inline src/views/Browser.vue:328]: I suggest you use flatMap here and Object.entries.

Something like:

```javascript
Object.entries(state.api.error).flatMap(([key, error]) => key === 'datasetsAnnptationsByCategory' ? Object.values(error) : [error])
```

**Review by arteymix — CHANGES_REQUESTED** [review]

File: src/views/Browser.vue (hunk 3 above)
Comment (libby-natola) [reply src/views/Browser.vue]: changed in commit c4eafa767ccb33eeb9bd48b16973fc6e16901bdf

**Review by arteymix — APPROVED** [review]

**Pull request merged**